crates/paimon/src/file_index/file_index_format.rs

```diff
@@ -0,0 +1,514 @@
+            let column_name_len = buffer.get_u16_le();
+            current_offset += 2;
+
+            // Column Name (variable-length UTF-8 string)
+            buffer = self
+                .reader
+                .read(current_offset..current_offset + column_name_len as u64)
+                .await?;
+            let column_name =
+                String::from_utf8(buffer.to_vec()).map_err(|e| Error::FileIndexFormatInvalid {
+                    message: format!("Invalid UTF-8 sequence in column name: {}", e),
+                })?;
+            current_offset += column_name_len as u64;
+
+            // Index Number (4 bytes)
+            buffer = self.reader.read(current_offset..current_offset + 4).await?;
+            let index_number = buffer.get_i32_le();
+            current_offset += 4;
+
+            let mut index_info_map = HashMap::new();
+            for _ in 0..index_number {
+                // Index Name Length (2 bytes)
+                buffer = self.reader.read(current_offset..current_offset + 2).await?;
+                let index_name_len = buffer.get_u16_le();
+                current_offset += 2;
+
+                // Index Name (variable-length UTF-8 string)
+                buffer = self
+                    .reader
+                    .read(current_offset..current_offset + index_name_len as u64)
+                    .await?;
+                let index_name = String::from_utf8(buffer.to_vec()).unwrap();
+                current_offset += index_name_len as u64;
+
+                // Start Pos (4 bytes)
+                buffer = self.reader.read(current_offset..current_offset + 4).await?;
+                let start_pos = buffer.get_i32_le();
+                current_offset += 4;
+
+                // Length (4 bytes)
+                buffer = self.reader.read(current_offset..current_offset + 4).await?;
+                let length = buffer.get_i32_le();
+                current_offset += 4;
+
+                index_info_map.insert(index_name, IndexInfo { start_pos, length });
+            }
+
+            self.header.insert(column_name, index_info_map);
+        }
+
+        buffer = self.reader.read(current_offset..current_offset + 4).await?;
+        let redundant_length = buffer.get_i32_le() as u64;
+        current_offset += 4;
+
+        if redundant_length > 0 {
+            let redundant_bytes = self
+                .reader
+                .read(current_offset..current_offset + redundant_length)
+                .await?;
+
+            if redundant_bytes.len() as u64 != redundant_length {
+                return Err(Error::FileIndexFormatInvalid {
+                    message: format!(
+                        "Expected to read {} redundant bytes, but found only {}",
+                        redundant_length,
+                        redundant_bytes.len()
+                    ),
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn read_column_index(
```

Comment:
I suggest splitting the column index logic out of the reader. For example, have the reader return a `FileIndex` struct that includes all the indexes, allowing users to use it to `get_column_index` or `get_index`.

Reply:
OK